demos/imagenet_classification_demo/grid_mat.hpp

```diff
@@ -0,0 +1,170 @@
+        int minCellSize = std::min(maxDisp.width / size.width, maxDisp.height / size.height);
+        cellSize = cv::Size(minCellSize, minCellSize);
+
+        for (int i = 0; i < size.width; i++) {
+            for (int j = 0; j < size.height; j++) {
+                points.emplace_back(cellSize.width * i, presenter.graphSize.height + cellSize.height * j);
+            }
+        }
+
+        outImg.create((cellSize.height * size.height) + presenter.graphSize.height,
+                       cellSize.width * size.width, CV_8UC3);
+        outImg.setTo(0);
+
+        fontType = cv::FONT_HERSHEY_PLAIN;
+        fontScale = 1.5;
+        thickness = 2;
+        textSize = cv::getTextSize("", fontType, fontScale, thickness, &baseline);
+        accuracyMessageSize = cv::getTextSize("Accuracy (top 0): 0.000", fontType, fontScale, thickness, &baseline);
+        testMessage = "Testing, please wait...";
+        testMessageSize = cv::getTextSize(testMessage, fontType, fontScale, thickness, &baseline);
+    }
+
+    void textUpdate(double avgFPS, double avgLatency, double accuracy,
+                    bool isFpsTest, bool showAccuracy,
+                    Presenter& presenter) {
+        rectangle(outImg,
+                  {0, 0}, {outImg.cols, presenter.graphSize.height},
+                  cv::Scalar(0, 0, 0), cv::FILLED);
+
+        presenter.drawGraphs(outImg);
+        
+        cv::Scalar textColor = cv::Scalar(255, 255, 255);
+        int textPadding = 10;
+
+        cv::putText(outImg,
+                    cv::format("FPS: %0.01f", avgFPS),
+                    cv::Point(textPadding, textSize.height + textPadding),
+                    fontType, fontScale, textColor, thickness);
+        cv::putText(outImg,
+                    cv::format("Latency: %dms", static_cast<int>(avgLatency * 1000)),
+                    cv::Point(textPadding, (textSize.height + textPadding) * 2),
+                    fontType, fontScale, textColor, thickness);
+        
+        if (showAccuracy) {
+            cv::putText(outImg,
+                        cv::format("Accuracy (top %d): %.3f", FLAGS_nt, accuracy),
+                        cv::Point(outImg.cols - accuracyMessageSize.width - textPadding, textSize.height + textPadding),
+                        fontType, fontScale, textColor, thickness);
+        }
+
+        if (isFpsTest) {
+            cv::putText(outImg,
+                        testMessage,
+                        cv::Point(outImg.cols - testMessageSize.width - textPadding,
+                                  (textSize.height + textPadding) * 2),
+                        fontType, fontScale, cv::Scalar(50, 50, 255), thickness);
+        }
+    }
+
+    void updateMat(ImageInfoList& imageInfos) {
+        ImageInfoList imagesToDraw;
+        if (!imageInfos.empty()) {
+            imagesToDraw.splice(imagesToDraw.end(), imageInfos);
+        }
+
+        if (!prevImgs.empty()) {
+            int prevSourceId = currSourceId - FLAGS_b;
```

Comment:
```suggestion
            int prevSourceId = currSourceId - prevImgs.size();
```
Then the wrapping `if` will not be needed.